operator/pkg/listener/watcher_events.go

```diff
@@ -0,0 +1,147 @@
+package listener
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kyma-project/kyma-operator/operator/api/v1alpha1"
+	"github.com/kyma-project/kyma-operator/operator/pkg/labels"
+	"github.com/kyma-project/kyma-operator/operator/pkg/release"
+	"github.com/kyma-project/kyma-operator/operator/pkg/util"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type WatcherEventsHandler struct {
+	client.Client
+	logger logr.Logger
+}
+
+func (h *WatcherEventsHandler) ProcessWatcherEvent(ctx context.Context) func(event.GenericEvent, workqueue.RateLimitingInterface) {
+	h.logger = log.FromContext(ctx).WithName("skr-watcher-events-processing")
+	return func(genericEvent event.GenericEvent, q workqueue.RateLimitingInterface) {
+		//Label component CR with a current timestamp
+		//when an event is received from the SKR watcher
+		componentName := genericEvent.Object.GetName()
+		clusterName := genericEvent.Object.GetClusterName()
+		h.logger.WithValues(
+			"component", componentName,
+			"cluster-name", clusterName,
+		).Info("started dispatching event")
+
+		kymaCR, err := h.getKymaCR(ctx, clusterName)
+		if err != nil {
+			h.logger.WithValues(
+				"component", componentName,
+				"cluster-name", clusterName,
+			).Error(err, "failed to get Kyma CR for component")
+			return
+		}
+
+		componentCR, err := h.getComponentCR(ctx, kymaCR, componentName, clusterName, q)
+		if err != nil {
+			h.logger.WithValues(
+				"component", componentName,
+				"cluster-name", clusterName,
+			).Error(err, "failed to get component CR")
+			return
+		}
+		err = h.updateComponentCRLabels(ctx, componentCR, kymaCR, q)
+		if err != nil {
+			h.logger.WithValues(
+				"component", componentName,
+				"cluster-name", clusterName,
+			).Error(err, "could not trigger the reconciliation loop")
+			return
+		}
+
+		h.logger.WithValues(
+			"component", componentName,
+			"cluster-name", clusterName,
+		).Info("successfully triggered component reconciliation loop")
+	}
+}
+
+func (h *WatcherEventsHandler) updateComponentCRLabels(ctx context.Context, componentCR *unstructured.Unstructured, kymaCR *v1alpha1.Kyma, q workqueue.RateLimitingInterface) error {
+	componentLabels := componentCR.GetLabels()
+	labelValue := strings.ReplaceAll(time.Now().UTC().Format(time.RFC3339), ":", ".")
+	componentLabels[labels.ListenerLastUpdated] = labelValue
+	componentCR.SetLabels(componentLabels)
+	err := h.Update(ctx, componentCR, &client.UpdateOptions{})
```

Comment:
I think we should never update a component CR Label through a Change Watcher with the K8s API. I think it could make more sense to instead issue a Kubernetes Event or set an Annotation instead. But even this is dangerous since many requests can lead to overloading the api server with label updates.

Reply:
The updating of the label has nothing do to with what we want to achieve, this is just a very temporary thing, to see that the listener is able to interact with the CR. In the end the listener should NOT set any label, it should patch the Component CR instead.

Reply:
As mentioned above. The new design approach resolves the above concerns!